src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -339,9 +332,12 @@ class ProtobufPlugin implements Plugin<Project> {
         // This cannot be called once task execution has started.
         variant.registerJavaGeneratingTask(
             generateProtoTask.get(), generateProtoTask.get().getOutputSourceDirectories())
-        linkGenerateProtoTasksToTaskName(
-            Utils.getKotlinAndroidCompileTaskName(project, variant.name),
-            sourceDirectorySetForGenerateProtoTask(variant.name, generateProtoTask))
+        project.plugins.withId("org.jetbrains.kotlin.android") {
```

Comment:
Looks like not need to be inside of the `postConfigure` block. I'm not sure, just looks so.

Reply:
I agree with you, but something in Kotlin (or Android) is going on and loading it eagerly I believe.

See "Keep Kotlin registration in postConfigure; it doesn't work otherwise" commit in https://github.com/google/protobuf-gradle-plugin/pull/614 where I abandoned moving it. I didn't fully look into it, but it definitely fails outside the `postConfigure` block and works within.